processor.go

```diff
@@ -113,7 +114,7 @@ func (p *ServiceProcessor) NewProtocol(tn *TreeNodeInstance, conf *GenericConfig
 // ProcessClientRequest takes a request from a websocket client, calculates the reply
 // and sends it back. It uses the path to find the appropriate handler-
 // function. It implements the Server interface.
-func (p *ServiceProcessor) ProcessClientRequest(path string, buf []byte) ([]byte, error) {
+func (p *ServiceProcessor) ProcessClientRequest(req *http.Request, path string, buf []byte) ([]byte, error) {
```

Comment:
Feels like `path` could and perhaps should be removed ? Redundand information.

Reply:
No, because the path has already been trimmed by the caller (to trim the service name) off the rfront. That would make all implementors of ProcessClientRequest responsible for doing this trimming, and maybe they don't even know the service name that they have been called for.

Reply:
I agree the information has been pre-processed but still if you look at it from a fresh point of view, you just see two arguments that are redundant. You can translate that to "I don't really know which one I should use if I want to code a new service." 
What do you think about having a wrapper around `http.Request` that can deliver the path with a simple method ?